src/OrbitBase/TypedefTest.cpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <mutex>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "OrbitBase/Typedef.h"
+
+namespace {
+struct TestTag {};
+
+struct Integer {
+  [[nodiscard]] Integer Add(const Integer& other) const { return {value + other.value}; }
+  int value{};
+};
+
+struct A {
+  int value{};
+};
+
+struct B : public A {};
+
+struct C {
+  int value{};
+  explicit C(const A& a) : value(a.value) {}
+};
+
+struct D {
+  int value{};
+  explicit D(A&& a) : value(std::move(a.value)) {}
+  D(const A& a) = delete;
+};
+
+}  // namespace
+
+static int Add(int i, int j) { return i + j; }
+
+namespace orbit_base {
+template <typename T>
+using Wrapper = Typedef<TestTag, T>;
+
+TEST(TypedefTest, CanInstantiate) {
+  const int kConstInt = 1;
+  Wrapper<int> wrapper_of_const(kConstInt);
+  EXPECT_EQ(*wrapper_of_const, kConstInt);
+
+  constexpr int kConstexprInt = 1;
+  Wrapper<int> wrapper_of_constexpr(kConstexprInt);
+  EXPECT_EQ(*wrapper_of_constexpr, kConstexprInt);
+
+  int non_const = 1;
+  Wrapper<int> wrapper_of_non_const(non_const);
+  EXPECT_EQ(*wrapper_of_non_const, non_const);
+
+  Wrapper<int> wrapper_of_literal(1);
+  EXPECT_EQ(*wrapper_of_literal, 1);
+
+  Wrapper<std::string> wrapper_of_string_literal("foo");
```

Comment:
Nit: Once it's wrapped it's not a string literal anymore. It's just a string which was initialized from a literal